kanidmd/src/lib/value.rs

```diff
@@ -1101,6 +1129,10 @@ impl Value {
             DbValueV1::DT(s) => PartialValue::new_datetime_s(&s)
                 .ok_or(())
                 .map(|pv| Value { pv, data: None }),
+            DbValueV1::EM(DbValueEmailAddressV1 { d }) => Ok(Value {
```

Comment:
What does the `DbValueV1` represent exactly? And what are `PartialValue`s? Also I think we should make make two constructors for `Value`, since the `data` field seems to be `None` most of the time. So we could have constructors like `Value::dataless` that only takes the `pv` field, and `Value::with_data` that takes both fields.

Reply:
As before, db varue is how this is serialised down. Value is in memory.

Reply:
Ah makes sense. What are your thoughts on the two `Value` initializers I proposed?

Also, I recall you saying somewhere that we used short variable names like `d` to reduce packet size when sending information. Can't we still use explicit, longer names and use `serde` macros the serialize them into shorter variants?

Reply:
I think the value thing is actually going to come up in https://github.com/kanidm/kanidm/issues/468 so let's move that talk there. And second we can use the longer names + serde macros, but again - william of the past didn't know this existed at the time. So my advice is open a new issue and just fix it. I know it's really tempting to want to clean up everything we see immediately in one code review, but we have to also consider a focus of this issue. It can quickly go from "add mail attributes" to "rewrite everything". So I'm not opposed to this change, but please open an issue so we can focus on it appropriately there :)

Reply:
@QnnOkabayashi Can you open the issue fro dbvalue serde renames?

Reply:
Okay, I opened it at https://github.com/kanidm/kanidm/issues/473